aiohttp_xmlrpc/handler.py

```diff
@@ -38,13 +38,32 @@ def __new__(cls, clsname, superclasses, attributedict):
         argmapping = getattr(instance, mapping_key)
         allowed_methods = getattr(instance, allowed_key)
 
+        # Try to get the METHOD_NAMES dictionnary
+        overwritten_names = getattr(instance, "METHOD_NAMES", {})
+
         for key in attributedict.keys():
             if not key.startswith(instance.METHOD_PREFIX):
                 continue
 
+            # Get the value of the corresponding function
             value = getattr(instance, key)
-            method_name = key.replace(instance.METHOD_PREFIX, "", 1)
+
+            # Check if the method is listed in METHOD_NAMES dict
+            if key in overwritten_names.values():
```

Comment:
I was thinking that it is possible to refactor this code to the base class is a metaclass. So in this case the mapping will be created on the class creation step, it's fast and allow renames of the methods using a decorator.

Reply:
I guess I didn't quite make myself clear. Now I'll try to make an example.

Reply:
Something like this

```diff
diff --git a/aiohttp_xmlrpc/handler.py b/aiohttp_xmlrpc/handler.py
index 734b824..5b22800 100644
--- a/aiohttp_xmlrpc/handler.py
+++ b/aiohttp_xmlrpc/handler.py
@@ -43,7 +43,11 @@ class XMLRPCViewMeta(ABCMeta):
                 continue
 
             value = getattr(instance, key)
-            method_name = key.replace(instance.METHOD_PREFIX, "", 1)
+
+            method_name = getattr(value, "__xmlrpc_name__", None)
+            if method_name is None:
+                method_name = key.replace(instance.METHOD_PREFIX, "", 1)
+
             allowed_methods[method_name] = key
             argmapping[method_name] = inspect.getfullargspec(value)
 
@@ -185,3 +189,10 @@ class XMLRPCView(View, metaclass=XMLRPCViewMeta):
             encoding="utf-8",
             pretty_print=cls.DEBUG,
         )
+
+
+def rename(new_name):
+    def decorator(func):
+        func.__xmlrpc_name__ = new_name
+        return func
+    return decorator
diff --git a/tests/test_handler.py b/tests/test_handler.py
index 60f6916..bb5532f 100644
--- a/tests/test_handler.py
+++ b/tests/test_handler.py
@@ -8,6 +8,7 @@ from aiohttp_xmlrpc.exceptions import ApplicationError
 from lxml import etree
 from lxml.builder import E
 
+from aiohttp_xmlrpc.handler import rename
 
 pytest_plugins = (
     "aiohttp.pytest_plugin",
@@ -52,6 +53,10 @@ class XMLRPCMain(handler.XMLRPCView):
     def rpc_dict_kw_only_args(self, d, *, foo, **kw):
         return (d, foo, kw)
 
+    @rename("method_with.new_name")
+    def rpc_ranamed(self):
+        return "renamed_function"
+
 
 class XMLRPCChild(XMLRPCMain):
     def rpc_child(self):
@@ -214,3 +219,8 @@ async def test_13_kw_only_args(client):
         {"foo": "bar"}, foo=32, spam="egg"
     )
     assert result == [{"foo": "bar"}, 32, {"spam": "egg"}]
+
+
+async def test_14_method_renaming(client):
+    result = await client.method_with.new_name()
+    assert result == "renamed_function"
```

Reply:
I will happy if you rebase your branch with my branch `hotfix/method-renaming` which contains this changes.

Reply:
Yes, indeed, decorators seem to be a better option than the first one.
I also thought about that but couldn't find how to implement it...

I will rebase my branch and rework it a bit later today!

Reply:
I've rebased and fixed my branch, it should be good now.